packages/cdk8s-plus-17/src/container.ts

```diff
@@ -508,6 +518,70 @@ export enum MountPropagation {
   BIDIRECTIONAL = 'Bidirectional',
 }
 
+/**
+ * Properties for creating ResourceRequirements
+ */
+export interface ResourceRequirementsProps {
+  readonly cpu?: IResourceRequirement<cdk8s.Cpu>;
```

Comment:
We avoid using generics in public interfaces because they don't translate well to `jsii`. Create separate types.